mmengine/hooks/checkpoint_hook.py

```diff
@@ -294,19 +294,19 @@ def after_val_epoch(self, runner, metrics):
             runner (Runner): The runner of the training process.
             metrics (dict): Evaluation results of all metrics
         """
+        if len(metrics) == 0:
+            runner.logger.warning(
+                'Since `metrics` is an empty dict, the behavior to save '
+                'the best checkpoint will be skipped in this evaluation.')
+            return
+
         self._save_best_checkpoint(runner, metrics)
 
     def _get_metric_score(self, metrics, key_indicator):
```

Comment:
Should we consider the `metrics is None` here? If it is necessary, `after_val_epoch` should also check if it is `None`. Besides,  it seems `_get_metric_score` is simply equivalent to `metrics[key_indicator]`

Reply:
In what case would it be None?

Reply:
`_get_metric_score` has been removed.